diff -crB -x .git AtmosphereO/stratosphere/fs_mitm/source/fsmitm_main.cpp Atmosphere/stratosphere/fs_mitm/source/fsmitm_main.cpp
*** AtmosphereO/stratosphere/fs_mitm/source/fsmitm_main.cpp	2018-09-21 18:56:22.363637200 -0700
--- Atmosphere/stratosphere/fs_mitm/source/fsmitm_main.cpp	2018-09-21 18:57:02.634409900 -0700
***************
*** 81,99 ****
      if (R_FAILED(rc))  {
          fatalSimple(0xCAFE << 4 | 3);
      }
-     
-     /* Check for exosphere API compatibility. */
-     u64 exosphere_cfg;
-     if (R_SUCCEEDED(splGetConfig((SplConfigItem)65000, &exosphere_cfg))) {
-         /* MitM requires Atmosphere API 0.1. */
-         u16 api_version = (exosphere_cfg >> 16) & 0xFFFF;
-         if (api_version < 0x0001) {
-             fatalSimple(0xCAFE << 4 | 0xFE);
-         }
-     } else {
-         fatalSimple(0xCAFE << 4 | 0xFF);
-     }
-     
      //splExit();
  }
  
--- 81,86 ----
diff -crB -x .git AtmosphereO/stratosphere/loader/source/ldr_main.cpp Atmosphere/stratosphere/loader/source/ldr_main.cpp
*** AtmosphereO/stratosphere/loader/source/ldr_main.cpp	2018-09-21 18:56:22.368637600 -0700
--- Atmosphere/stratosphere/loader/source/ldr_main.cpp	2018-09-21 19:01:06.562688600 -0700
***************
*** 84,97 ****
      if (R_FAILED(rc))  {
          fatalSimple(0xCAFE << 4 | 3);
      }
-     
-     /* Check for exosphere API compatibility. */
-     u64 exosphere_cfg;
-     if (R_FAILED(splGetConfig((SplConfigItem)65000, &exosphere_cfg))) {
-         //fatalSimple(0xCAFE << 4 | 0xFF);
-         /* TODO: Does Loader need to know about target firmware/master key revision? If so, extract from exosphere_cfg. */
-     }
-     
      //splExit();
  }
  
--- 84,89 ----
diff -crB -x .git AtmosphereO/stratosphere/pm/source/pm_main.cpp Atmosphere/stratosphere/pm/source/pm_main.cpp
*** AtmosphereO/stratosphere/pm/source/pm_main.cpp	2018-09-21 18:56:22.373637100 -0700
--- Atmosphere/stratosphere/pm/source/pm_main.cpp	2018-09-21 19:00:12.177294100 -0700
***************
*** 93,105 ****
      if (R_FAILED(rc))  {
          fatalSimple(0xCAFE << 4 | 6);
      }
-     
-     /* Check for exosphere API compatibility. */
-     u64 exosphere_cfg;
-     if (R_FAILED(splGetConfig((SplConfigItem)65000, &exosphere_cfg))) {
-         fatalSimple(0xCAFE << 4 | 0xFF);
-         /* TODO: Does PM need to know about target firmware/master key revision? If so, extract from exosphere_cfg. */
-     }
  }
  
  void __appExit(void) {
--- 93,98 ----
