diff -crB -x .git Atmosphere/stratosphere/fs_mitm/source/fsmitm_main.cpp AtmoPatch/stratosphere/fs_mitm/source/fsmitm_main.cpp
*** Atmosphere/stratosphere/fs_mitm/source/fsmitm_main.cpp	2018-09-18 08:57:40.342663718 +0300
--- AtmoPatch/stratosphere/fs_mitm/source/fsmitm_main.cpp	2018-09-20 14:02:09.418675866 +0300
***************
*** 82,99 ****
          fatalSimple(0xCAFE << 4 | 3);
      }
      
-     /* Check for exosphere API compatibility. */
-     u64 exosphere_cfg;
-     if (R_SUCCEEDED(splGetConfig((SplConfigItem)65000, &exosphere_cfg))) {
-         /* MitM requires Atmosphere API 0.1. */
-         u16 api_version = (exosphere_cfg >> 16) & 0xFFFF;
-         if (api_version < 0x0001) {
-             fatalSimple(0xCAFE << 4 | 0xFE);
-         }
-     } else {
-         fatalSimple(0xCAFE << 4 | 0xFF);
-     }
-     
      //splExit();
  }
  
--- 82,87 ----
diff -crB -x .git Atmosphere/stratosphere/loader/source/ldr_main.cpp AtmoPatch/stratosphere/loader/source/ldr_main.cpp
*** Atmosphere/stratosphere/loader/source/ldr_main.cpp	2018-09-18 08:57:40.346663748 +0300
--- AtmoPatch/stratosphere/loader/source/ldr_main.cpp	2018-09-20 14:02:31.546614271 +0300
***************
*** 85,97 ****
          fatalSimple(0xCAFE << 4 | 3);
      }
      
-     /* Check for exosphere API compatibility. */
-     u64 exosphere_cfg;
-     if (R_FAILED(splGetConfig((SplConfigItem)65000, &exosphere_cfg))) {
-         //fatalSimple(0xCAFE << 4 | 0xFF);
-         /* TODO: Does Loader need to know about target firmware/master key revision? If so, extract from exosphere_cfg. */
-     }
-     
      //splExit();
  }
  
--- 85,90 ----
diff -crB -x .git Atmosphere/stratosphere/loader/source/ldr_npdm.cpp AtmoPatch/stratosphere/loader/source/ldr_npdm.cpp
*** Atmosphere/stratosphere/loader/source/ldr_npdm.cpp	2018-09-20 14:01:15.686839803 +0300
--- AtmoPatch/stratosphere/loader/source/ldr_npdm.cpp	2018-09-20 14:04:04.410389520 +0300
***************
*** 53,68 ****
  
  
  FILE *NpdmUtils::OpenNpdm(u64 title_id) {
!     if (ContentManagement::ShouldOverrideContents()) {
!         if (ContentManagement::ShouldReplaceWithHBL(title_id)) {
!             return OpenNpdmFromHBL();
          }
!         FILE *f_out = OpenNpdmFromSdCard(title_id);
!         if (f_out != NULL) {
!             return f_out;
          }
      }
-     return OpenNpdmFromExeFS();
  }
  
  Result NpdmUtils::LoadNpdmInternal(FILE *f_npdm, NpdmUtils::NpdmCache *cache) {
--- 53,91 ----
  
  
  FILE *NpdmUtils::OpenNpdm(u64 title_id) {
!     if (title_id == 0x010000000000100D) {
!         Result rc;
!         rc = hidInitialize();
!         if (R_FAILED(rc)){
!             fatalSimple(MAKERESULT(Module_Libnx, LibnxError_InitFail_HID));
          }
!         hidScanInput();
!         u64 kDown = hidKeysDown(CONTROLLER_P1_AUTO);
!         if((kDown & KEY_R) == 0) {
!           hidExit();
!           FILE *f_out = OpenNpdmFromSdCard(title_id);
!           if (f_out != NULL) {
!            return f_out;
          }
+           return OpenNpdmFromExeFS();
+         }
+         else {
+          hidExit();
+          return OpenNpdmFromExeFS();
+         }
+     }
+     else {
+         if (ContentManagement::ShouldOverrideContents()) {
+             if (ContentManagement::ShouldReplaceWithHBL(title_id)) {
+                 return OpenNpdmFromHBL();
+             }
+             FILE *f_out = OpenNpdmFromSdCard(title_id);
+             if (f_out != NULL) {
+                 return f_out;
+             }
+         }
+         return OpenNpdmFromExeFS();
      }
  }
  
  Result NpdmUtils::LoadNpdmInternal(FILE *f_npdm, NpdmUtils::NpdmCache *cache) {
diff -crB -x .git Atmosphere/stratosphere/loader/source/ldr_nso.cpp AtmoPatch/stratosphere/loader/source/ldr_nso.cpp
*** Atmosphere/stratosphere/loader/source/ldr_nso.cpp	2018-09-20 14:01:15.686839803 +0300
--- AtmoPatch/stratosphere/loader/source/ldr_nso.cpp	2018-09-20 14:05:01.790275015 +0300
***************
*** 61,78 ****
  }
  
  FILE *NsoUtils::OpenNso(unsigned int index, u64 title_id) {
!     if (ContentManagement::ShouldOverrideContents()) {
!         if (ContentManagement::ShouldReplaceWithHBL(title_id)) {
!             return OpenNsoFromHBL(index);
          }
!         FILE *f_out = OpenNsoFromSdCard(index, title_id);
!         if (f_out != NULL) {
!             return f_out;
!         } else if (CheckNsoStubbed(index, title_id)) {
!             return NULL;
          }
      }
-     return OpenNsoFromExeFS(index);
  }
  
  bool NsoUtils::IsNsoPresent(unsigned int index) {
--- 61,103 ----
  }
  
  FILE *NsoUtils::OpenNso(unsigned int index, u64 title_id) {
!     if (title_id == 0x010000000000100D) {
!         Result rc;
!         rc = hidInitialize();
!         if (R_FAILED(rc)){
!             fatalSimple(MAKERESULT(Module_Libnx, LibnxError_InitFail_HID));
          }
!         hidScanInput();
!         u64 kDown = hidKeysDown(CONTROLLER_P1_AUTO);
!         if((kDown & KEY_R) == 0) {
!               hidExit();
!               FILE *f_out = OpenNsoFromSdCard(index, title_id);
!               if (f_out != NULL) {
!                 return f_out;
!               } else if (CheckNsoStubbed(index, title_id)) {
!                return NULL;
!               } else {
!                return OpenNsoFromExeFS(index);
!               }
          }
+         else {
+             hidExit();
+             return OpenNsoFromExeFS(index); }
+     }
+     else {
+         if (ContentManagement::ShouldOverrideContents()) {
+             if (ContentManagement::ShouldReplaceWithHBL(title_id)) {
+                 return OpenNsoFromHBL(index);
+             }
+             FILE *f_out = OpenNsoFromSdCard(index, title_id);
+             if (f_out != NULL) {
+                 return f_out;
+             } else if (CheckNsoStubbed(index, title_id)) {
+                 return NULL;
+             }
+         }
+         return OpenNsoFromExeFS(index);
      }
  }
  
  bool NsoUtils::IsNsoPresent(unsigned int index) {
diff -crB -x .git Atmosphere/stratosphere/pm/source/pm_main.cpp AtmoPatch/stratosphere/pm/source/pm_main.cpp
*** Atmosphere/stratosphere/pm/source/pm_main.cpp	2018-09-18 08:57:40.346663748 +0300
--- AtmoPatch/stratosphere/pm/source/pm_main.cpp	2018-09-20 14:05:20.042242083 +0300
***************
*** 93,105 ****
      if (R_FAILED(rc))  {
          fatalSimple(0xCAFE << 4 | 6);
      }
-     
-     /* Check for exosphere API compatibility. */
-     u64 exosphere_cfg;
-     if (R_FAILED(splGetConfig((SplConfigItem)65000, &exosphere_cfg))) {
-         fatalSimple(0xCAFE << 4 | 0xFF);
-         /* TODO: Does PM need to know about target firmware/master key revision? If so, extract from exosphere_cfg. */
-     }
  }
  
  void __appExit(void) {
--- 93,98 ----
